src/responses/src/sql.rs

```diff
@@ -0,0 +1,44 @@
+/*!
+Response types for a [sql request](https://www.elastic.co/guide/en/elasticsearch/reference/current/sql-rest.html)
+*/
+
+use parsing::IsOkOnSuccess;
+use serde_json::Value;
+
+/** Response for a [sql request][sql-request]. */
+#[derive(Deserialize, Debug)]
+pub struct SqlResponse {
+    columns: Vec<SqlColumn>,
+    rows: Vec<Vec<Value>>,
+}
+
+impl SqlResponse {
+    /** Gets a reference to the result columns. */
+    pub fn columns(&self) -> &Vec<SqlColumn> {
```

Comment:
These could be `&[SqlColumn]`

Reply:
Fixed. Should they `.rows()` type be something other than `&Vec<Vec<Value>>` as well?